testsuite/classlibrary/TestObject.sc

```diff
@@ -0,0 +1,7 @@
+TestObject : UnitTest {
+
+	test_writeDefFile_emptyName {
+		this.assertException({ nil.writeDefFile("") }, Error, "nil.writeDefFile(\"\") should fail");
```

Comment:
this test should use an `Object`, not `nil`. if for some reason Nil.writeDefFile is implemented in the future this will be testing the wrong thing.

Reply:
I tried `Object().writeDefFile("")` but that fails differently. 
<details>

```supercollider
ERROR: Message 'writeDef' not understood.
RECEIVER:
Instance of Object {    (0x1209cc898, gc=B8, fmt=00, flg=00, set=00)
  instance variables [0]
}
ARGS:
Instance of File {    (0x11ff23298, gc=B8, fmt=00, flg=00, set=02)
  instance variables [1]
    fileptr : nil
}

PROTECTED CALL STACK:
	Meta_MethodError:new	0x11b0b7140
		arg this = DoesNotUnderstandError
		arg what = nil
		arg receiver = an Object
	Meta_DoesNotUnderstandError:new	0x11b0b9100
		arg this = DoesNotUnderstandError
		arg receiver = an Object
		arg selector = writeDef
		arg args = [ a File ]
	Object:doesNotUnderstand	0x1168ad2c0
		arg this = an Object
		arg selector = writeDef
		arg args = nil
	ArrayedCollection:do	0x11e397340
		arg this = [ an Object ]
		arg function = a Function
		var i = 0
	Collection:writeDef	0x11e37cb00
		arg this = [ an Object ]
		arg file = a File
	Function:prTry	0x11d25ff00
		arg this = a Function
		var result = nil
		var thread = a Thread
		var next = nil
		var wasInProtectedFunc = false
	
CALL STACK:
	DoesNotUnderstandError:reportError
		arg this = <instance of DoesNotUnderstandError>
	Nil:handleError
		arg this = nil
		arg error = <instance of DoesNotUnderstandError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of DoesNotUnderstandError>
	Object:throw
		arg this = <instance of DoesNotUnderstandError>
	Function:protect
		arg this = <instance of Function>
		arg handler = <instance of Function>
		var result = <instance of DoesNotUnderstandError>
	Function:doOnStartUp
		arg this = <instance of Function>
	Meta_StartUp:defer
		arg this = <instance of Meta_StartUp>
		arg object = <instance of Function>
	Object:writeDefFile
		arg this = <instance of Object>
		arg name = "/Users/marcin/Library/Applic..."
		arg dir = "/Users/marcin/Library/Applic..."
		arg overwrite = true
	Interpreter:interpretPrintCmdLine
		arg this = <instance of Interpreter>
		var res = nil
		var func = <instance of Function>
		var code = "Object().writeDefFile("")"
		var doc = nil
		var ideClass = <instance of Meta_ScIDE>
	Process:interpretPrintCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Message 'writeDef' not understood.
RECEIVER: an Object
```
</details>

and `TestObject.run` gives
<details>

```
-> TestObject
RUNNING UNIT TEST 'TestObject'
PASS: a TestObject: test_writeDefFile_emptyName - Object().writeDefFile("") should fail
Received exception of class 'DoesNotUnderstandError', with message: 'ERROR: Message 'writeDef' not understood.'

UNIT TESTS FOR 'TestObject' COMPLETED
There were no failures
```
</details>
I'd appreciate if you pointed me to the way to write this test. I'm just trying to get a minor fix in :)

Reply:
oh huh, that's a bit surprising, but i guess this is some sort of undocumented interface design. sorry, i didn't know. in that case, what you have is good, but could you leave a comment indicating that we don't test Object().writeDefFile directly because it's not fully implemented?

Reply:
Thanks! I added the comment, not sure if it should also have "FIXME" prefix or is it okay as it.